### DISCUSSION TITLE
Preserving Individual Filenames When Doing Batch Cropping

### DISCUSSION THREAD

**Discussion by DReaper** [ImageMagick/ImageMagick#3028] [Help]
I want to split an image into 4x4. However, I want to do this over many images. I need the output file to share the same filename as the file it was working off of, while starting a new count for each cropping.

I have this command:

magick convert p%03d.jpg[1-100] -crop 25%x25% +repage p%d_crop_%d.jpg

I know the output name is incorrect. For example, I want the output name to be p001_crop_0.jpg to p001_crop_15.jpg, then p002_crop_0.jpg to p002_crop_15.jpg when it works off the second image. I want the crop count to reset when it goes to a new image. What command would I have to do for this? 

As well, is it possible for the crop count to start from 1, instead of 0? Thank you.

**Comment by snibgo** [comment]
We can do this:
```
magick -verbose p%03d.jpg[1-100] -crop 25%x25% +repage -scene 4 -set filename:A %t %[filename:A]_%d.jpg
```
That gives us flenames like p001_4.jpg etc, and "4" increments with each file, but it doesn't reset with each new input JPEG file.

To do what you want, I think you need a script, with a shell "for" loop to process each input file separately.
